src/model_tuner/model_tuner_utils.py

```diff
@@ -947,20 +947,27 @@ def train_val_test_split(
         #     X = X.join(self.dropped_strat_cols)
         # Determine the stratify parameter based on stratify and stratify_cols
 
-        if stratify_cols and stratify_y:
+        if stratify_cols is not None and stratify_y:
             # Creating stratification columns out of stratify_cols list
-            stratify_key = pd.concat([X[stratify_cols], y], axis=1)
-        elif stratify_cols:
+            if type(stratify_cols) == pd.DataFrame:
+                stratify_key = pd.concat([stratify_cols, y], axis=1)
```

Comment:
Added support for stratify_cols to be a dataframe is useful need to make sure this is valid